x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotAllocator.java

```diff
@@ -136,7 +133,8 @@ public void allocateUnassigned(
                 unassignedAllocationHandler.initialize(
                     allocateUnassignedDecision.getTargetNode().getId(),
                     allocateUnassignedDecision.getAllocationId(),
-                    allocation.snapshotShardSizeInfo().getShardSize(shardRouting, ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE),
+                    allocation.snapshotShardSizeInfo()
```

Comment:
I am not sure this makes sense for replicas, since the `InternalSnapshotsInfoService` only keeps the snapshot size for as long as the primary is recovering. This could need refinement for sure, possibly by taking the primary's actual size or similar, but even for regular replicas, this is most likely just `UNAVAILABLE_EXPECTED_SHARD_SIZE` too. Perhaps we should just use:

```
                        final long shardSize = DiskThresholdDecider.getExpectedShardSize(shard,
                            ShardRouting.UNAVAILABLE_EXPECTED_SHARD_SIZE,
                            allocation.clusterInfo(), allocation.snapshotShardSizeInfo(), allocation.metadata(), allocation.routingTable());
```

here like it used to be when delegating the `BalancedShardsAllocator`?

Reply:
My thinking was that since `getExpectedShardSize` will effectively always return `UNAVAILABLE_EXPECTED_SHARD_SIZE` the solution here will at least work out the correct size while the primary is recovering while that method will never give us anything but `UNAVAILABLE_EXPECTED_SHARD_SIZE` but I guess it's a pointless corner case => went with your suggestion now to keep things simple :)